src/service/viewport/viewport-impl.js

```diff
@@ -554,19 +554,15 @@ export class Viewport {
         offset = 0;
         break;
     }
-    let newScrollTop;
-    let curScrollTopPromise;
 
-    if (this.useLayers_) {
-      newScrollTop = elementRect.top + offset;
-      curScrollTopPromise = Promise.resolve(0);
-    } else {
-      const calculatedScrollTop = elementRect.top - this.paddingTop_ + offset;
-      newScrollTop = Math.max(0, calculatedScrollTop);
-      curScrollTopPromise = this.getElementScrollTop_(parent);
-    }
-
-    return curScrollTopPromise.then(curScrollTop => {
+    return this.getElementScrollTop_(parent).then(curScrollTop => {
+      let newScrollTop;
+      if (this.useLayers_) {
+        newScrollTop = elementRect.top + offset + curScrollTop;
+      } else {
+        const calculatedScrollTop = elementRect.top - this.paddingTop_ + offset;
+        newScrollTop = Math.max(0, calculatedScrollTop);
```

Comment:
We should do this same `Math.max` for Layers as well. Shouldn't happen in practice, but better to be safe.

Reply:
Added.